src/utils/gitUtils.ts

```diff
@@ -165,6 +165,20 @@ export async function promptForDefaultBranch(context: IActionContext, repo: Repo
     }
 }
 
+export async function gitPull(repo: Repository): Promise<void> {
+    try {
+        const options: ProgressOptions = {
+            location: ProgressLocation.Notification,
+            title: localize('executingGitPull', 'Executing "git pull"...')
+        };
+        await window.withProgress(options, async () => {
+            await repo.pull();
```

Comment:
I would prefer you only wrap this line in the try/catch.  As pointed out in this PR: https://github.com/microsoft/vscode-azurestaticwebapps/pull/369#discussion_r625468725, we aren't verifying that the error is actually a Git error.  I'll fix that, but I still want to be careful where we use `GitError`.